kernel/async/monitor-common/async/Place.cc

```diff
@@ -29,15 +29,17 @@
 namespace mythos {
 namespace async {
 
-  Place places[BOOT_MAX_THREADS];
-  CoreLocal<Place*> localPlace_ KERNEL_CLM_HOT; // for DeployHWThread
+  CoreLocal<Place*> localPlace_ KERNEL_CLM_HOT;
+  Place places[MYTHOS_MAX_THREADS];
 
-  void Place::init(size_t apicID)
+  void Place::init(cpu::ThreadID threadID, cpu::ApicID apicID)
   {
-    MLOG_INFO(mlog::async, "init Place", DVAR(this), DVAR(apicID));
+    MLOG_INFO(mlog::async, "init Place", DVAR(this), DVAR(threadID), DVAR(apicID));
+    localPlace_.setAt(threadID, this);
+    this->threadID = threadID;
     this->apicID = apicID;
     this->nestingMonitor = true;
-    this->_cr3 = PhysPtr<void>(cpu::getPageTable());
+    this->_cr3 = PhysPtr<void>(cpu::getPageTable()); /// @todo likely wrong and wrong place
```

Comment:
This just initializes a copy of the cr3 register. It is used for 1) avoiding a TLB shotdown if setting a value that has not changed, and 2) for helping the PML4 deletion broadcast. Initializing it here is not strictly necessary, since cr3 should be the static kernel PML4 it this point. However, it is nice to keep the copy consistent.